From 2281fa50f6e13278d3fd4f028377b8249b7b3cdb Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Fri, 8 Sep 2017 09:06:54 +0200
Subject: [PATCH] src/corelib/arch/qatomic_arm.h: fix build on ARMv8 32-bit

The Qt build fails on ARMv8 32-bit because it falls back to using the
ARMv5 code that uses the swp instruction, which no longer exists:

/tmp/ccSxVIzw.s: Assembler messages:
/tmp/ccSxVIzw.s:127: Error: swp{b} use is obsoleted for ARMv8 and later
/tmp/ccSxVIzw.s:190: Error: swp{b} use is obsoleted for ARMv8 and later

So, this commit adjusts the ARM atomic code to assume ARMv8 is similar
to ARMv7.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 src/corelib/arch/qatomic_arm.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/corelib/arch/qatomic_arm.h b/src/corelib/arch/qatomic_arm.h
index 77cd66f272..e011929d3e 100644
--- a/src/corelib/arch/qatomic_arm.h
+++ b/src/corelib/arch/qatomic_arm.h
@@ -47,7 +47,8 @@ QT_BEGIN_HEADER
 #if defined(__ARM_ARCH_7__) \
     || defined(__ARM_ARCH_7A__) \
     || defined(__ARM_ARCH_7R__) \
-    || defined(__ARM_ARCH_7M__)
+    || defined(__ARM_ARCH_7M__) \
+    || defined(__ARM_ARCH_8A__)
 # define QT_ARCH_ARMV7
 QT_BEGIN_INCLUDE_HEADER
 # include "QtCore/qatomic_armv7.h"
-- 
2.13.5

